File: client-clojure/src/sirqul_io_t_platform/specs/leaderboard_response.clj
(ns sirqul-io-t-platform.specs.leaderboard-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.asset-response :refer :all]
            [sirqul-io-t-platform.specs.asset-response :refer :all]
            [sirqul-io-t-platform.specs.application-mini-response :refer :all]
            )
  (:import (java.io File)))


(def leaderboard-response-data
  {
   (ds/opt :leaderboardId) int?
   (ds/opt :rankType) string?
   (ds/opt :rankMode) string?
   (ds/opt :sortType) string?
   (ds/opt :limitation) int?
   (ds/opt :title) string?
   (ds/opt :description) string?
   (ds/opt :metaData) string?
   (ds/opt :iconAsset) asset-response-spec
   (ds/opt :bannerAsset) asset-response-spec
   (ds/opt :active) boolean?
   (ds/opt :application) application-mini-response-spec
   })

(def leaderboard-response-spec
  (ds/spec
    {:name ::leaderboard-response
     :spec leaderboard-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/LeaderboardResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import LeaderboardResponse._

case class LeaderboardResponse (
  leaderboardId: Option[Long],
rankType: Option[String],
rankMode: Option[String],
sortType: Option[String],
limitation: Option[Integer],
title: Option[String],
description: Option[String],
metaData: Option[String],
iconAsset: Option[AssetResponse],
bannerAsset: Option[AssetResponse],
active: Option[Boolean],
application: Option[ApplicationMiniResponse])

object LeaderboardResponse {
  import DateTimeCodecs._

  implicit val LeaderboardResponseCodecJson: CodecJson[LeaderboardResponse] = CodecJson.derive[LeaderboardResponse]
  implicit val LeaderboardResponseDecoder: EntityDecoder[LeaderboardResponse] = jsonOf[LeaderboardResponse]
  implicit val LeaderboardResponseEncoder: EntityEncoder[LeaderboardResponse] = jsonEncoderOf[LeaderboardResponse]
}
